os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -544,6 +472,81 @@ static int loading_thread(int argc, char *argv[])
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+/****************************************************************************
+ * Name: binary_manager_read_header
+ *
+ * Description:
+ *	 This function reads header and checks whether it is valid or not.
+ *
+ ****************************************************************************/
+int binary_manager_read_header(char *path, binary_header_t *header_data)
+{
+	int fd;
+	int ret;
+	int read_size;
+	int file_size;
+	bool need_unlink;
+	uint32_t crc_hash = 0;
+	uint32_t check_crc = 0;
+	uint8_t crc_buffer[CRC_BUFFER_SIZE];
+
+	memset(header_data, 0, sizeof(binary_header_t));
+
+	need_unlink = false;
+
+	fd = open(path, O_RDONLY);
+	if (fd < 0) {
+		bmdbg("Failed to open %s: %d, errno %d\n", path, ret, errno);
+		return ERROR;
+	}
+
+	/* Read the binary header */
+	ret = read(fd, (FAR uint8_t *)header_data, sizeof(binary_header_t));
+	if (ret != sizeof(binary_header_t)) {
+		bmdbg("Failed to read %s: %d, errno %d\n", path, ret, errno);
+		goto errout_with_fd;
+	}
+
+	/* Verify header data */
+	if (header_data->bin_type != BIN_TYPE_ELF) {
+		need_unlink = true;
+		bmdbg("Invalid header data : headersize %d, binsize %d, ramsize %d, bintype %d\n", header_data->header_size, header_data->bin_size, header_data->bin_ramsize, header_data->bin_type);
+		goto errout_with_fd;
+	}
+
+	/* Caculate checksum and Verify it */
+	check_crc = crc32part((uint8_t *)header_data + CHECKSUM_SIZE, header_data->header_size, check_crc);
+	file_size = header_data->bin_size;
+	while (file_size > 0) {
+		read_size = file_size < CRC_BUFFER_SIZE ? file_size : CRC_BUFFER_SIZE;
+		ret = read(fd, (FAR uint8_t *)crc_buffer, read_size);
+		if (ret < 0 || ret != read_size) {
+			bmdbg("Failed to read : %d, errno %d\n", ret, errno);
```

Comment:
Is it okay not to unlink the path for this case?

Reply:
Yes. I think it should be removed when it is invalid only.